lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1336,7 +1337,8 @@ assignScriptRedeemers pparams ti utxo redeemers tx =
             (Map Alonzo.RdmrPtr (Either ErrAssignRedeemers Alonzo.ExUnits))
     evaluateExecutionUnitsAlonzo indexedRedeemers alonzoTx = do
         let pparams' = Cardano.toLedgerPParams
-                Cardano.ShelleyBasedEraAlonzo pparams
+                Cardano.ShelleyBasedEraAlonzo
+                    (Cardano.unbundleProtocolParams pparams)
```

Comment:
Here we should call `unbundleLedgerShelleyBasedProtocolParams` instead of `toLedgerPParams . unbundleProtocolParams` to actually leverage the `BundledProtocolParameters` abstraction and avoid costly conversions